Sources/MapboxNavigation/CarPlayManager.swift

```diff
@@ -496,36 +497,27 @@ extension CarPlayManager: CPInterfaceControllerDelegate {
             carPlayMapViewController.recenterButton.isHidden = true
         }
         
-        if let userInfo = template.userInfo as? Dictionary<String, Any>,
-           let currentActivity = userInfo[CarPlayManager.currentActivityKey] as? CarPlayActivity {
+        if let currentActivity = template.currentActivity {
             self.currentActivity = currentActivity
         } else {
             self.currentActivity = nil
         }
+        
+        if self.currentActivity == .browsing {
+            navigationMapView?.navigationCamera.follow()
+        }
     }
     
     public func templateDidAppear(_ template: CPTemplate, animated: Bool) {
         delegate?.carPlayManager(self, templateDidAppear: template, animated: animated)
         
         guard interfaceController?.topTemplate == mainMapTemplate,
-              template == interfaceController?.rootTemplate,
-              let carPlayMapViewController = carPlayMapViewController else { return }
-        
-        let navigationMapView = carPlayMapViewController.navigationMapView
-        navigationMapView.removeRoutes()
-        navigationMapView.removeContinuousAlternativesRoutes()
-        navigationMapView.removeWaypoints()
+              template == interfaceController?.rootTemplate else { return }
+        self.removeRoutesFromMap()
     }
     
     public func templateWillDisappear(_ template: CPTemplate, animated: Bool) {
         delegate?.carPlayManager(self, templateWillDisappear: template, animated: animated)
-        
```

Comment:
The following mode of camera should happen in the `templateWillAppear` and in the `.browsing` mode.